src/experimental/tools/mediaCapabilitiesProber/probers/HDCPPolicy.ts

```diff
@@ -63,41 +56,32 @@ export default function probeHDCPPolicy(
     ],
   };
 
-  return eme
-    .requestMediaKeySystemAccess(keySystem, [drmConfig])
-    .then((mediaKeysSystemAccess) => {
-      return mediaKeysSystemAccess
-        .createMediaKeys()
-        .then((mediaKeys) => {
-          if (!("getStatusForPolicy" in mediaKeys)) {
-            // do the check here, as mediaKeys can be either be native MediaKeys or
-            // custom MediaKeys from compat.
-            throw new Error(
-              "MediaCapabilitiesProber >>> API_CALL: " +
-                "getStatusForPolicy API not available",
-            );
-          }
-          return (
-            mediaKeys as {
-              getStatusForPolicy: (policy: {
-                minHdcpVersion: string;
-              }) => Promise<IMediaKeyStatus>;
-            }
-          )
-            .getStatusForPolicy(policy)
-            .then((result: IMediaKeyStatus) => {
-              let status: [ProberStatus];
-              if (result === "usable") {
-                status = [ProberStatus.Supported];
-              } else {
-                status = [ProberStatus.NotSupported];
-              }
-              return status;
-            });
-        })
-        .catch(() => {
-          const status: [ProberStatus] = [ProberStatus.Unknown];
-          return status;
-        });
-    });
+  const mediaKeysSystemAccess = await eme.requestMediaKeySystemAccess(keySystem, [
+    drmConfig,
+  ]);
+
+  const mediaKeys = await mediaKeysSystemAccess.createMediaKeys();
+  try {
+    if (!("getStatusForPolicy" in mediaKeys)) {
+      // do the check here, as mediaKeys can be either be native MediaKeys or
+      // custom MediaKeys from compat.
+      throw new Error("getStatusForPolicy API not available");
+    }
+  } catch (err) {
+    const error = err instanceof Error ? err : new Error("Unknown Error");
+    log.error("MCP: `probeHDCPPolicy` didn't succeed to create a MediaKeys", error);
```

Comment:
From what I read, the error log here is not accurate:
We can get here if the API `getStatusForPolicy` does not exist, not because the MediaKeys failed to be created.

Reply:
You're right `requestMediaKeySystemAccess` and `createMediaKeys` don't even have their try/catch.

I'll fix that